sources/Adapters/picoTracker/gui/picoTrackerEventManager.cpp

```diff
@@ -54,15 +54,20 @@ int picoTrackerEventManager::MainLoop() {
       queue->Empty(); // Avoid duplicates redraw
       redrawing_ = false;
     }
+
+    if (loops == (3 * 1000000)) {
 #ifdef PICOSTATS
-    if (loops == 100000) {
       printf("Usage %.1f% CPU\n", ((float)events / loops) * 100);
+      // measure_freqs();
+      measure_free_mem();
+#endif
       events = 0;
       loops = 0;
-      //      measure_freqs();
-      measure_free_mem();
+      event = new picoTrackerEvent();
```

Comment:
What about we move the move the event creation logic into the timer handler, if you want a 1Hz timer we can do an `if (gTime % 1000) { ...}` and instead of process the event, which would be unpredictable and potentially last more than 1ms, we would just add it to the queue, and let the main loop handle it.
There is also the matter of that heap allocation, but we can take care of that later.

Reply:
ah cool thanks yes this sounds like a better was to do it, will do!